one-pager-maker/src/pages/Edit.tsx

```diff
@@ -1,54 +1,100 @@
-import { useState } from 'react'
-import Markdown from 'react-markdown'
 import './Edit.css'
+import { useEffect, useState } from 'react'
+import Markdown from 'react-markdown'
+import { useUpdateDocumentMutation, useFetchDocumentQuery } from "../redux/document/documentsApi.ts";
+import { useNavigate, useParams } from "react-router-dom";
+import { Document, Status } from "../redux/document/documentType.ts";
+import { auth } from '../firebase';
 
 function Edit() {
-  const [markdown, setMarkdown] = useState('');
-  const toMarkdownText = (e:React.ChangeEvent<HTMLTextAreaElement>) => {
-    setMarkdown(e.target.value);
-  };
+  const uid = auth.currentUser?.uid ?? "";
+
+  const [title, setTitle] = useState('');
+  const [contents, setContents] = useState('');
+  const [status, setStatus] = useState<Status>('draft');
+  const [contributors, setContributors] = useState('');
+  const [reviewers, setReviewers] = useState('');
+  const [document, setDocument] = useState<Document | undefined>(undefined);
+  const urlParams = useParams<{ id: string }>()
+  const navigate = useNavigate();
+  const {data} = useFetchDocumentQuery({uid: uid, docId: urlParams.id ?? ""});
+  const [updateDocument] = useUpdateDocumentMutation();
+  
+  useEffect(() => {
+    console.log(data);
+    if (data === undefined) return;
+    setDocument(data);
+    setTitle(data.title);
+    setContents(data.contents);
+    setStatus(data.status);
+    setContributors(data.contributors.join(','));
+    setReviewers(data.reviewers.join(','));
+  }, [data]);
+
+  const onChangeTitle = (e:React.ChangeEvent<HTMLInputElement>) => setTitle(e.target.value);
+  const onChangeContents = (e:React.ChangeEvent<HTMLTextAreaElement>) => setContents(e.target.value);
+  const onChangeStatus = (e:React.ChangeEvent<HTMLSelectElement>) => setStatus(e.target.value as Status);
+  const onChangeContributors =  (e:React.ChangeEvent<HTMLInputElement>) => setContributors(e.target.value);
+  const onChangeReviewers =  (e:React.ChangeEvent<HTMLInputElement>) => setReviewers(e.target.value);
+  const onClickSave = async () => {
+    try {
+      await updateDocument({
+          uid: uid,
+          documentData: {
+              id:urlParams?.id ?? "",
+              title: title,
+              contents: contents,
+              status: status,
+              owner_id: uid,
+              contributors: contributors.split(','),
+              reviewers: reviewers.split(','),
```

Comment:
It might be better to handle `contributors `and `reviewers` as an `array` in UI Component, rather than a `string`.
But this code is okay for now, do refactoring after merged.